all: enable staticcheck unused

keegancsmith opened the pull request:
I cleaned this up a while ago, but then forgot to enable it so we ended up regressing in a few places. See individual commits, I'll cc the relevant author in to confirm.

keegancsmith commented on enterprise/internal/campaigns/syncer.go:
```diff
@@ -388,23 +388,6 @@ func (s *ChangesetSyncer) GroupChangesetsBySource(ctx context.Context, cs ...*ca
-	}
-
-	return all, err
-}
```
cc @ryanslade

ryanslade replied:
Happy for this to be removed, we use `ListChangesetSyncData` instead

keegancsmith commented on cmd/repo-updater/repos/bitbucketserver.go:
```diff
@@ -476,7 +475,3 @@ func (s *BitbucketServerSource) listAllLabeledRepos(ctx context.Context, label s
-
-func unixMilliToTime(ms int64) time.Time {
-	return time.Unix(0, ms*int64(time.Millisecond))
-}
```
cc @unknwon

unknwon replied:
I don't think I wrote this code but removing it LGTM :)

mrnugget replied:
I wrote it. It got lost in a refactor.

keegancsmith commented on enterprise/cmd/repo-updater/authz/request_queue.go:
```diff
@@ -24,8 +24,7 @@ type requestType int
-	requestTypeRepo
+	requestTypeRepo requestType = iota + 1
 	requestTypeUser
 )
```
cc @unknwon

unknwon replied:
LGTM

unknwon commented:
> I cleaned this up a while ago, but then forgot to enable it so we ended up regressing in a few places. See individual commits, I'll cc the relevant author in to confirm.

IIRC @rvantonder disabled this few weeks back because of its flaky on unused variables (or something false-positive).

keegancsmith commented:
> > I cleaned this up a while ago, but then forgot to enable it so we ended up regressing in a few places. See individual commits, I'll cc the relevant author in to confirm.
> 
> IIRC @rvantonder disabled this few weeks back because of its flaky on unused variables (or something false-positive).

aah ok. Well we updated the version of staticcheck we use so lets see if it starts firing again :)
